reana_server/reana_admin.py

```diff
@@ -214,39 +218,102 @@ def import_users(ctx, admin_access_token, file_):
 @admin_access_token_option
 @click.option(
     '--id',
+    'id_',
     help='The id of the user.')
 @click.option(
     '-e',
     '--email',
     help='The email of the user.')
-def token_grant(id, email, admin_access_token):
-    """."""
-    user_id = 'dc4aef21-c324-4fd5-8480-c724de1f0c06'
-    user_email = 'john.doe@google.com'
-    user_granted_token = 'c0fa47fa00ae4013a13fd7n'
-    # create a token in UserToken table and activate it
-    # create an audit log entry
-    # send notification to user by email
-    click.secho(f'Token for user {user_id} ({user_email}) granted.\n'
-                f'\nToken: {user_granted_token}', fg='green')
+@click.option(
+    '-f',
+    '--force',
+    is_flag=True,
+    default=False,
+    help='Grant token even if user has a revoked access token.')
+def token_grant(admin_access_token, id_, email, force):
+    """Grant a token to the selected user."""
+    try:
+        admin = User.query.filter_by(id_=ADMIN_USER_ID).one_or_none()
+        if admin_access_token != admin.access_token:
+            raise ValueError('Admin access token invalid.')
+        user = _get_user_by_criteria(id_, email)
+        error_msg = None
+        if not user:
+            error_msg = f'User {id_ or email} does not exist.'
+        elif user.access_token:
+            error_msg = (f'User {user.id_} ({user.email}) has already an'
+                         ' active access token.')
+        elif (not force and user.access_token_status in
+              [UserTokenStatus.revoked.name, None]):
+            error_msg = (f'User {user.id_} ({user.email}) access token status'
+                         f' is {user.access_token_status}, if you want to'
+                         ' proceed in any case use --force option.')
+        if error_msg:
+            click.secho(f'ERROR: {error_msg}', fg='red')
+            sys.exit(1)
+
+        user_granted_token = secrets.token_urlsafe(16)
```

Comment:
We have the logic to create tokens a bit spread, it would be nice to centralise.
```console
$ rg "token_urlsafe\(16\)"
reana-server/tests/test_cli.py
44:    user_access_token = secrets.token_urlsafe(16)

reana-server/reana_server/reana_admin.py
255:        user_granted_token = secrets.token_urlsafe(16)

reana-server/reana_server/utils.py
85:            user_access_token = secrets.token_urlsafe(16)
```
(Not for now, but we can open an issue after merging this)

Reply:
Definitely.